addons/portal/static/src/js/portal_share_document.js

```diff
@@ -0,0 +1,110 @@
+odoo.define('portal.share_document', function (require) {
+'use strict';
+
+var core = require('web.core');
+var data_manager = require('web.data_manager');
+var SystrayMenu = require('web.SystrayMenu');
+var WebClient = require('web.WebClient');
+var Widget = require('web.Widget');
+
+var _t = core._t;
+
+var ProtalShareDoc = Widget.extend({
+    template: 'portal.sharing_icon',
+    events: {
+        "click": "_onClick",
+    },
+    xmlDependencies: ['/portal/static/src/xml/portal_share_document.xml'],
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Update the system tray icon for share document, based on view type hides/shows icon
+     * Show Share Icon only if 1) action context has share_icon True 2) form view of portal.mixin inherited object
+     */
+    update: function (tag, descriptor, widget) {
+        var self = this;
+        if (widget && tag === 'action') {
+            this._controller = widget;
+            this.shareAction = descriptor.context.share_action ? descriptor.context.share_action : 'portal.mail_invite_share_action';
+            if (this._controller.viewType === 'form') {
+                if (widget.activeActions && widget.activeActions.share) {
+                    this.$el.removeClass('o_hidden');
+                } else {
+                    this.$el.addClass('o_hidden');
+                }
+            } else {
+                this.getSession().user_has_group('base.group_erp_manager').then(function (has_group) {
```

Comment:
you should try to change this to have to make the rpc only once, not everytime a view is opened.  Remember, a rpc is costly, we want to minimize them.